### PR TITLE
Update AttestationRecord and CrosslinkRecord Proto

### PR DESCRIPTION
**Pull request by prestonvanloon**
To match spec at https://github.com/ethereum/eth2.0-specs/commit/11d44738635d89a3d0e28f194d12fa2fcca80a81

~Awaiting confirmation on a question regarding distinct data types for PendingAttestationRecord vs AttestationRecord.~

### CODE REVIEW

File: proto/beacon/p2p/v1/types.proto [hunk 1]
```diff
@@ -79,11 +79,19 @@ message CandidatePoWReceiptRootRecord {
   uint64 votes = 2;
 }
 
+// PendingAttestationRecord is deprecated entirely by AttestationRecord.
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto]: Let me confirm that this is true. I don't see why this is a different data structure.

Comment (terenc3t) [reply proto/beacon/p2p/v1/types.proto]: We should have 2. One is `AggregatedAttestation` which represents `AttestationRecord` in spec
Another one is `PendingAttestationRecord`
`AggregatedAttestation` tracks attestations before processing. `PendingAttestationRecord` tracks attestations post processing

**Comment by prestonvanloon** [comment]
Consensus is that we should not have optional fields and therefore these must be distinct data types. I’ll update shortly

**Comment by rauljordan** [comment]
Just a small conflict to fix, looks good

**Pull request merged**